### ISSUE TITLE
False positives with type variable shadow warning in 0.10.5

### ISSUE THREAD

**Issue by garyb** [purescript/purescript#2542]
This function now raises a warning about a shadowed `a` (the `updateSelect` part, the rest is highlighted to illustrate there's no other `a` in scope):

https://github.com/slamdata/slamdata/blob/5988934c04e9b52902d74b9e1e7b685bc9d3664f/src/SlamData/Workspace/Card/BuildChart/Funnel/Component.purs#L234-L250

**Comment by LiamGoodacre** [comment]
Minimal test case:
```purescript
type T = forall a. Array a
foo :: T
foo = bar where
  bar :: forall a. Array a
  bar = []
```

**Comment by LiamGoodacre** [comment]
For the same reason, the following compiles but shouldn't:
```purescript
type T = forall a. Array a
foo :: T
foo = bar where
  bar :: Array a
  bar = []
```

**Comment by LiamGoodacre** [comment]
Probably worth adding both of those as test cases.  The former being a `warning` test that expects no warnings.

**Comment by garyb** [comment]
Did you gain any clue to why this might have started happening since 0.10.4? At first I was thinking it was synonym-expanding related, but that went in in .4, so it's a bit puzzling to me at least! 😕

**Comment by LiamGoodacre** [comment]
I stopped looking when Phil self-assigned - @paf31 are you looking into this?

**Comment by paf31** [comment]
I was going to look at this today. I know where the issue is - it was caused by the fix I put into 0.10.5, when I moved synonym expansion before the kind check.
